src/js/index.js

```diff
@@ -36,33 +36,15 @@ function tippy(targets, options, one) {
     polyfillElementPrototypeProperties(targets)
   }
 
-  const references = getArrayOfElements(targets)
-  const firstReference = references[0]
-
-  const instances = (one && firstReference
-    ? [firstReference]
-    : references
-  ).reduce((acc, reference) => {
-    const tip = reference && createTippy(reference, props)
-    if (tip) {
-      acc.push(tip)
+  const instances = getArrayOfElements(targets).reduce((acc, reference) => {
+    const instance = reference && createTippy(reference, props)
+    if (instance) {
+      acc.push(instance)
     }
     return acc
   }, [])
 
-  const collection = {
-    targets,
-    props,
-    instances,
-    destroyAll() {
-      collection.instances.forEach(instance => {
-        instance.destroy()
-      })
-      collection.instances = []
-    },
-  }
-
-  return collection
+  return isSingular(targets) ? instances[0] : instances
```

Comment:
Isn't that the other way around? If it is a singular then it can't have `0`-indexed item in it

Reply:
It's always an array, if it's singular it's an array with one item in it, so we return it directly

Reply:
`isSingular` checks whether the value is a plain object or instance of `Element`. Are you sure it is working as intended?

Reply:
Ah sorry, for some reason I was thinking it was checking an array to be singular. My mistake

Reply:
Hah yes, all of the tests would be failing since it relies on `tippy()` returning correctly 😉